pkg/obc/obc.go

```diff
@@ -435,3 +506,63 @@ func CheckPhase(obc *nbv1.ObjectBucketClaim) {
 		log.Printf("⏳ OBC %q Phase is %q", obc.Name, obc.Status.Phase)
 	}
 }
+
+// GenerateAccountKeys regenerate noobaa OBC account S3 keys
+func GenerateAccountKeys(name string, accountName string) error {
+	log := util.Logger()
+
+	if accountName == "" {
+		log.Fatalf(`❌  account name cannot be empty.\n`)
+	}
+
+	var accessKeys nb.S3AccessKeys
+
+	sysClient, err := system.Connect(true)
+	if err != nil {
+		return err
+	}
+
+	// Checking that we can find the secret before we are calling the RPC to change the credentials.
+	secret := util.KubeObject(bundle.File_deploy_internal_secret_empty_yaml).(*corev1.Secret)
+	secret.Namespace = options.Namespace
+	secret.Name = name
+	if !util.KubeCheckQuiet(secret) {
+		log.Fatalf(`❌  Could not find secret: %s, will not regenerate keys.`, secret.Name)
+	}
+
+	if !util.KubeCheckQuiet(secret) {
+		log.Fatalf(`❌  Could not find secret: %s, will not regenerate keys.`, secret.Name)
+	}
+
+	err = sysClient.NBClient.GenerateAccountKeysAPI(nb.GenerateAccountKeysParams{
+		Email:	accountName,
+	})
+	if err != nil {
+		return err
+	}
+
+	// GenerateAccountKeysAPI have no replay so we need to read the account in order to get the new credentials
+	accountInfo, err := sysClient.NBClient.ReadAccountAPI(nb.ReadAccountParams{
+		Email: accountName,
+	})
+	if err != nil {
+		return err
+	}
+ 
+	accessKeys = accountInfo.AccessKeys[0]
+
+	secret.StringData = map[string]string{}
+	secret.StringData["AWS_ACCESS_KEY_ID"] = accessKeys.AccessKey
+	secret.StringData["AWS_SECRET_ACCESS_KEY"] = accessKeys.SecretKey
+
+	//If we will not be able to update the secret we will print the credentials as they allready been changed by the RPC
+	if !util.KubeUpdate(secret) {
```

Comment:
@liranmauda  @dannyzaken  can we handle this situation better? maybe somehow hang the failure on the OBC/OB or somehow get the bucket into a rejected state and ask the user to regenerate again?

Reply:
At this point, the account credentials are already changed and we are failing to update the secret.
The apps that are using this secret will also disconnect, and as we cant revert the best way to work around it is to regenerate again, and unless there is a fundamental issue with the secret that prevents us from updating it, it will fix it.
as a middle solution, we just print the new credentials (which can be manually edited into the secret as b64, by us/support/user).

@dannyzaken @romayalon